Cabal/Distribution/Utils/Generic.hs

```diff
@@ -166,72 +164,22 @@ writeFileAtomic targetPath content = do
 -- This is a modification of the UTF8 code from gtk2hs and the
 -- utf8-string package.
 
+{-# DEPRECATED fromUTF8 "Please use 'decodeStringUtf8', 'fromUTF8BS', or 'fromUTF8BS'" #-}
 fromUTF8 :: String -> String
-fromUTF8 []     = []
-fromUTF8 (c:cs)
-  | c <= '\x7F' = c : fromUTF8 cs
-  | c <= '\xBF' = replacementChar : fromUTF8 cs
-  | c <= '\xDF' = twoBytes c cs
-  | c <= '\xEF' = moreBytes 3 0x800     cs (ord c .&. 0xF)
-  | c <= '\xF7' = moreBytes 4 0x10000   cs (ord c .&. 0x7)
-  | c <= '\xFB' = moreBytes 5 0x200000  cs (ord c .&. 0x3)
-  | c <= '\xFD' = moreBytes 6 0x4000000 cs (ord c .&. 0x1)
-  | otherwise   = replacementChar : fromUTF8 cs
+fromUTF8 = decodeStringUtf8 . map c2w
```

Comment:
fwiw, we could also remove `to/fromUTF8` w/o any deprecation. Nothing in the code-base needs it anymore